    @Test(expected = IllegalArgumentException.class)
    public void add_invalidName_throwsIllegalArgumentException() throws Exception {
        // Arrange
        builder.setType(FakeTypes.Mine.MINENAMETYPE_TYPE);

        // Act
        try {
            builder.add(
                    "test_id",
                    "Sharlston Colliery",
                    new NameImpl(FakeTypes.Mine.MINE_NAMESPACE, "INVALID_NAME")); // Intentionally
            // invalid
            // name.
        } catch (IllegalArgumentException iae) {
            ExceptionChecker.assertExceptionMessage(
                    iae,
                    "Could not locate attribute: urn:org:example:INVALID_NAME in type: urn:org:example:MineNameType");
        }
    }

